### ISSUE TITLE
Improve documentation for pretokenized datasets

### ISSUE THREAD

**Issue by ein-ich** [axolotl-ai-cloud/axolotl#615]
### What piece of documentation is affected?

**How to use your custom pretokenized dataset**
-     Do not pass a type:
-     Dataset must contain input_ids, attention_mask, labels in columns

### What part(s) of the article would you like to see updated?

Change last line to
- Columns in Dataset must be exactly input_ids, attention_mask and labels

The documentation for pretokenized datasets seems sparse in general.
I added bos, eos, pad tokens and it does train, but I don't know if that is the correct way.
- Would "pad_to_sequence_len" add pad tokens with pretokenized datasets?
- I assume and hope it doesn't add additional bos or eos tokens.
- What's the effect of "sample_packing" in this context? Would it work when I don't manually add pad tokens?

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
> Change last line to
> Columns in Dataset must be exactly input_ids, attention_mask and labels

This sounds like a good change. Would you like to PR this?

> Would "pad_to_sequence_len" add pad tokens with pretokenized datasets?
> I assume and hope it doesn't add additional bos or eos tokens.

Pads during training. It should be pad tokens.

https://github.com/OpenAccess-AI-Collective/axolotl/blob/c1382e79b6eb599f7fd210fd1a621b97bfed0d51/src/axolotl/utils/trainer.py#L723-L730

> What's the effect of "sample_packing" in this context? Would it work when I don't manually add pad tokens?

Sample packing packs samples more efficiently. I think there was a bug when this is not on while using sample packing, hence, we have a config check when `cfg.sample_packing and not cfg.pad_to_sequence_len:`

**Comment by ein-ich** [comment]
> This sounds like a good change. Would you like to PR this?

Will do that.

> > I assume and hope it doesn't add additional bos or eos tokens.
>
> Pads during training. It should be pad tokens.

What I wanted to say is, I assume axolotl doesn't add any bos or eos tokens in general when using a pretokenized dataset (in addition to the pad tokens).

The loss looks better now that I don't manually add the pad tokens.
